### ISSUE TITLE
TransactionStmtFlowTest fails after error/typecheck change migration

### ISSUE THREAD

**Issue by Manuri** [ballerina-platform/ballerina-lang#11355]
**Description:**
Compilation fails with
```bash
org.ballerinalang.launcher.BLauncherException
	at org.ballerinalang.launcher.LauncherUtils.createLauncherException(LauncherUtils.java:223)
	at org.ballerinalang.launcher.LauncherUtils.getExecutableProgram(LauncherUtils.java:331)
	at org.ballerinalang.launcher.util.BCompileUtil.compile(BCompileUtil.java:306)
	at org.ballerinalang.launcher.util.BCompileUtil.compile(BCompileUtil.java:217)
	at org.ballerinalang.launcher.util.BCompileUtil.compile(BCompileUtil.java:199)
	at org.ballerinalang.launcher.util.BCompileUtil.compile(BCompileUtil.java:116)
	at org.ballerinalang.test.statements.transaction.TransactionStmtFlowTest.setup(TransactionStmtFlowTest.java:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:59)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:451)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:222)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:142)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:163)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:105)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1187)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1116)
	at org.testng.TestNG.runSuites(TestNG.java:1028)
	at org.testng.TestNG.run(TestNG.java:996)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)

```
Migration done with https://github.com/ballerina-platform/ballerina-lang/pull/11440

**Comment by Manuri** [comment]
This issue occurs when, below two records are in the given order in the transaction_stmt.bal. If the order is reversed, issue doesn't occur. Seems this is not related to any transaction related functionality.
```ballerina
public type TrxErrorData record {
     string message = "";
     error? cause = ();
     string data = "";
     !...
 };

 public type TrxError error<string, TrxErrorData>;
```

**Comment by SupunS** [comment]
Fixed with #11536
